codegen/src/test/java/org/web3j/codegen/unit/gen/java/Setup.java

```diff
@@ -54,13 +55,10 @@ public static void setUp() throws IOException, ClassNotFoundException {
                         "contracts",
                         "GreeterTest.java");
         classAsFile = new File(urlAsString);
-        File greeter = new File(urlAsString.substring(0, urlAsString.indexOf("org/")));
-        greeterContractClass =
-                new ClassProvider(greeter)
-                        .getClasses().stream()
-                                .filter(className -> className.getSimpleName().equals("Greeter"))
-                                .findAny()
-                                .orElse(null);
+        CompilerClassLoader compilerClassLoader =
+                new CompilerClassLoader(temp, Setup.class.getClassLoader().getResource("java/"));
+
+        greeterContractClass = compilerClassLoader.findClass("org.com.test.contract.Greeter");
```

Comment:
As mentioned before, use the method meant for this. 

Please read carefully the Javadoc before overriding a Java standard class internals.